src/aac/plugins/gen_json/gen_json_impl.py

```diff
@@ -10,22 +11,51 @@
 )
 from aac.validator import validation
 
+
 plugin_name = "gen-json"
 
 
-def print_json(architecture_files: list[str]) -> PluginExecutionResult:
-    """Print the parsed_models from the parsed architecture_files values in JSON format."""
+def print_json(architecture_files: list[str], output_directory: str = None) -> PluginExecutionResult:
+    """Prints the parsed_models from the parsed architecture_files values in JSON format.
+
+       (optional) if output directory is given to the cli, parsed_models with be writen to
+        a new file in user defined directory
+
+    """
+
+    def to_json(architecture_file: str, output_directory: str) -> str:
+        file_name = os.path.basename(architecture_file)
+        file_ext = os.path.splitext(file_name)[1]
 
-    def to_json(architecture_file: str) -> str:
         with validation(parse_file, architecture_file) as result:
-            return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
+
+            if output_directory is None:
+                return f"File: {architecture_file}\n{json.dumps(result.model, indent=2)}\n"
+
+            else:
+                if file_ext == '':
+                    file_name = (file_name + '.json')
+                else:
+                    file_name = file_name.replace(file_ext, '.json')
+
+                output_file_path = os.path.join(output_directory, file_name)
+                outFile = open(output_file_path, "w")
+                outFile.write(json.dumps(result.model, indent=2))
+                outFile.close()
+                return f"Successfully added {file_name} to directory"
```

Comment:
```suggestion
                return f"Successfully wrote JSON to {output_file_path}."
```

Reply:
Since there might be a possibility of a list of files being parsed, I would like to add the file name to this line as well. I have made a commit with my fix.

Reply:
The `file_name` will be included in the `output_file_path` - it'll just be an absolute file path rather than showing only the file name.

Reply:
okay, agreed and fixed with only output_file_path.